salt/gis.sls

```diff
@@ -0,0 +1,127 @@
+# get the software packages installed
+base:
+  pkg.installed:
+    - install_recommends: false
+    - pkgs:
+      # just some basic packages, not specific to the carto setup.
+      - git 
+      - unzip
+      - curl
+      - build-essential
+      - software-properties-common
+    - refresh: true
+
+# installing the required GIS-software, plus...
+# create the postgresql database and enabling PostGIS
+gis:
+  pkg.installed:
+    - install_recommends: false
+    - pkgs:
+      # here come the goodies; PostGIS is elementary to a current dev. environment.
+      - postgresql-9.3-postgis-2.1
+      - postgresql-contrib-9.3
+      - proj-bin
+      - libgeos-dev
+  # Let's make it easy for our GIS-in-aid software to access our PostGIS database
+  file.blockreplace:
+    - name: /etc/postgresql/9.3/main/pg_hba.conf
+    - marker_start: "# TYPE  DATABASE"
+    - marker_end: '# "local" is'
+    - content: "local   all             gisuser                                 trust"
```

Comment:
if you're connecting with a local socket from the same user you shouldn't need to modify `pg_hba.conf`